test/runner/e2e_update_rel_test.cpp

```diff
@@ -0,0 +1,165 @@
+#include "test_helper/test_helper.h"
+
+using namespace kuzu::testing;
+
+class UpdateRelTest : public DBTest {
+public:
+    string getInputCSVDir() override {
+        return TestHelper::appendKuzuRootPath("dataset/rel-insertion-tests/");
+    }
+    string getUpdateRelQuery(string srcTable, string dstTable, string relation, int64_t srcID,
+        int64_t dstID, string setPropertyClause) {
+        return StringUtils::string_format(
+            "MATCH (p1:%s)-[e:%s]->(p2:%s) WHERE p1.ID = %d AND p2.ID = %d " + setPropertyClause,
+            srcTable.c_str(), relation.c_str(), dstTable.c_str(), srcID, dstID);
+    }
+    string getInsertKnowsRelQuery(string srcTable, string dstTable, int64_t srcID, int64_t dstID) {
+        return StringUtils::string_format("MATCH (p1:%s), (p2:%s) WHERE p1.ID = %d AND p2.ID "
+                                          "= %d create (p1)-[:knows {length: %d}]->(p2);",
+            srcTable.c_str(), dstTable.c_str(), srcID, dstID, srcID);
+    }
+};
+
+TEST_F(UpdateRelTest, UpdateIntProp) {
```

Comment:
I'm confused about this test. This PR does not implement checkpointing, right? For example I don't see any changes to Lists::writeToInMemList(), which is what RelTable::prepareCommit...(...) calls. But you seem to be testing through a conn->query(), which I think will trigger a sequence of auto commits, that updates happen successfully. What am I missing? 

Or are these all working for now because you never empty the updates/modifications in RelUpdateStore and always read rel updates for both read and write transactions? If this is the case, there should be a note saying some of these behavior are temporary and a todo to fix them eventually.

Reply:
This PR doesn't do anything with checkpointing. If we update a list, the changes will be stored in the updateStore and we never clean our updateStore.
The reason why i did this is: 
Guodong was refactoring the lists read apis, and the readValues() api doesn't have access to transaction. I need to change the readValues() api. To avoid conflict with guodong (also i am not doing any transaction stuffs in this PR), i decided to make the updates visibile to all transactions (that's the reason why i don't have an if statement for `readUpdatesToPropertyVectorIfExists`), so i didn't need to change the read api of lists and i can still test the lists updates. Thoses tests will also be refactored (with some transaction tests) in the next pr, so i decided to make these tests simple for the current PR.

However, i think this may be confusing and guodong's PR has been merged. In the revised version of this PR, i added a `startWriteTrx` statement at the begining of each test.